hss/include/hssStatsPromClient.h

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright 2020-present Open Networking Foundation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef _INCLUDE_hssStats_H__
+#define _INCLUDE_hssStats_H__
+
+#include <prometheus/gauge.h>
+#include <prometheus/counter.h>
+#include <prometheus/exposer.h>
+#include <prometheus/registry.h>
+
+
+using namespace prometheus;
+extern std::shared_ptr<Registry> registry;
+
+
+void hssStatsSetupPrometheusThread(uint16_t port);
+
+enum class hssStatsCounter {
+	MME_MSG_TX_S6A_AUTH_INFO_ANSWER_SUCCESS,
+	MME_MSG_TX_S6A_AUTH_INFO_ANSWER_FAILURE,
+	MME_MSG_TX_S6A_UPDATE_LOCATION_ANSWER_SUCCESS,
+	MME_MSG_TX_S6A_UPDATE_LOCATION_ANSWER_FAILURE,
+	MME_MSG_TX_S6A_PURGE_ANSWER_SUCCESS,
+	MME_MSG_TX_S6A_PURGE_ANSWER_FAILURE,
+	MME_MSG_TX_S6A_CANCEL_LOCATION_REQUEST,
+	MME_MSG_RX_S6A_AUTH_INFO_REQUEST,
+	MME_MSG_RX_S6A_UPDATE_LOCATION_REQUEST,
+	MME_MSG_RX_S6A_PURGE_REQUEST
+};
+
+struct Node 
+{
+   hssStatsCounter id;
+	std::string label_k;
+	std::string label_v;
+
+	Node(hssStatsCounter id, std::string label_k, std::string label_v)
+	{
+		this->id = id;
+		this->label_k = label_k;
+		this->label_v = label_v;
+	}
+
+	// operator== is required to compare keys in case of hash collision
+	bool operator==(const Node &p) const
+	{
+		return label_k == p.label_k && label_v == p.label_v && id == p.id;
+	}
+};
+
+struct hash_fn
+{
+	std::size_t operator() (const Node &node) const
+	{
+		std::size_t h1 = std::hash<std::string>()(node.label_k);
+		std::size_t h2 = std::hash<std::string>()(node.label_v);
+		std::size_t h3 = std::size_t(node.id);
+		return h1 ^ h2 ^ h3;
+	}
+};
+
+
+class DynamicMetricObject {
```

Comment:
what is the use of this empty class.?

Reply:
This is encapsulation to cover other dynamic classes. Its inherited by other classes. and this is used to put in map.